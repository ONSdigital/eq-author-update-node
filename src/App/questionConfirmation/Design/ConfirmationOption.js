import PropTypes from "prop-types";
import React from "react";
import DummyMultipleChoice from "App/page/Design/answers/dummy/MultipleChoice";
import {
  StyledOption,
  Flex,
  OptionField,
} from "App/page/Design/answers/MultipleChoiceAnswer/Option";
import { Label } from "components/components-original/Forms";
import WrappingInput from "components/components-original/Forms/WrappingInput";

import { RADIO } from "constants/answer-types";
import { CONFIRMATION_OPTION } from "constants/validation-error-types";

import withValidationError from "enhancers/withValidationError";

export const UnwrappedConfirmationOption = ({
  value,
  name,
  label,
  onChange,
  onUpdate,
  getValidationError,
}) => (
  <StyledOption>
    <div>
      <Flex>
        <DummyMultipleChoice type={RADIO} />
        <OptionField>
          <Label htmlFor={`option-label-${name}`}>{label}</Label>
          <WrappingInput
            id={`option-label-${name}`}
            name={`${name}.label`}
            value={value.label}
            placeholder={""}
            onChange={onChange}
            onBlur={onUpdate}
            data-test={`${name}-option-label`}
            bold
            errorValidationMsg={getValidationError({
              field: "label",
              label: "Confirmation label",
              requiredMsg: `Enter ${label.toLowerCase()}`,
            })}
          />
        </OptionField>
      </Flex>
      <OptionField>
        <Label htmlFor={`option-description-${name}`}>
          Description (optional)
        </Label>
        <WrappingInput
          id={`option-description-${name}`}
          name={`${name}.description`}
          onChange={onChange}
          value={value.description}
          onBlur={onUpdate}
          data-test={`${name}-option-description`}
        />
      </OptionField>
    </div>
  </StyledOption>
);

UnwrappedConfirmationOption.propTypes = {
  value: PropTypes.shape({
    label: PropTypes.string,
    description: PropTypes.string,
  }).isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onUpdate: PropTypes.func.isRequired,
  getValidationError: PropTypes.func.isRequired,
};

export default withValidationError(CONFIRMATION_OPTION)(
  UnwrappedConfirmationOption
);
