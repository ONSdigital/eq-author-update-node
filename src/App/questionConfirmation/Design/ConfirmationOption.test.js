import React from "react";
import { shallow } from "enzyme";

import { UnwrappedConfirmationOption } from "./ConfirmationOption";

describe("ConfirmationOption", () => {
  it("should render", () => {
    const wrapper = shallow(
      <UnwrappedConfirmationOption
        name="option"
        value={{ label: "label", description: "description" }}
        label="Option label"
        onChange={jest.fn()}
        onUpdate={jest.fn()}
        getValidationError={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
