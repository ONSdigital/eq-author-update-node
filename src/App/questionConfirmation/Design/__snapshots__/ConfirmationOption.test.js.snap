// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationOption should render 1`] = `
<styled.div
  duration={200}
>
  <div>
    <styled.div>
      <styled.div
        type="Radio"
      />
      <Styled(Field)>
        <Label
          bold={true}
          htmlFor="option-label-option"
        >
          Option label
        </Label>
        <withChangeHandler(undefined)
          bold={true}
          data-test="option-option-label"
          id="option-label-option"
          name="option.label"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          placeholder=""
          value="label"
        />
      </Styled(Field)>
    </styled.div>
    <Styled(Field)>
      <Label
        bold={true}
        htmlFor="option-description-option"
      >
        Description (optional)
      </Label>
      <withChangeHandler(undefined)
        data-test="option-option-description"
        id="option-description-option"
        name="option.description"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        value="description"
      />
    </Styled(Field)>
  </div>
</styled.div>
`;
